src/v3/src/util/formUtils.ts

```diff
@@ -37,3 +54,214 @@ export const handleFormFieldBlur = (
+        const redirectUri = `${baseUrl}/sso/idps/${idpConfig.id}?stateToken=${stateHandle}`;
+        if (idpConfig.className) {
+          // @ts-expect-error OKTA-609464 - className missing from IdpConfig type
+          idp.className = idpConfig.className;
+        }
+        return {
+          name: IDX_STEP.REDIRECT_IDP,
+          type: idpConfig.type,
+          idp,
+          href: redirectUri,
+        };
+      });
+  }
+
+  return [];
+};
+
+// TO DO: OKTA-609477 - Add rest of supported social idp icons here
+const getIdpButtonIcon = (idpType: string) : string => {
+  const idpIconMapping: Record<string, string> = {
+    facebook: facebookIconSvg,
+    google: googleIconSvg,
+    linkedin: linkedinIconSvg,
+    microsoft: msIconSvg,
+    apple: appleIconSvg,
+    okta: oktaIconSvg,
+  };
+  return idpIconMapping[idpType];
+};
+
+/**
+ * Example of `redirect-idp` remediation.
+ * {
+ *   "name": "redirect-idp",
+ *   "type": "MICROSOFT",
+ *   "idp": {
+ *      "id": "0oa2szc1K1YPgz1pe0g4",
+ *      "name": "Microsoft IDP"
+ *    },
+ *   "href": "http://localhost:3000/sso/idps/0oa2szc1K1YPgz1pe0g4?stateToken=BB...AA",
+ *   "method": "GET"
+ * }
+ *
+ */
+export const getIdpButtonElements = (
+  transaction: IdxTransaction,
+  widgetProps: WidgetProps,
+) : ButtonElement[] => {
+  const { neededToProceed: remediations } = transaction;
+  let redirectIdpRemediations = remediations.filter(
+    (idp) => idp.name === IDX_STEP.REDIRECT_IDP,
+  ) || [];
+  // Add Idp buttons from widget config (openid-connect)
+  redirectIdpRemediations = redirectIdpRemediations.concat(
+    getConfigIdpButtonRemediations(transaction, widgetProps),
+  );
+
+  // create button elements from idp objects
+  const idpButtonElements = redirectIdpRemediations.map((idpObject) => {
+    let type = idpObject.type?.toLowerCase() || '';
+    let displayName;
+
+    if (IDP.SUPPORTED_SOCIAL_IDPS.includes(type) === false) {
+      type = 'general-idp';
+      // OKTA-396684 - makes sure that custom idps always have a name
+      displayName = loc('customauth.sign.in.with.label', 'login', [idpObject.idp?.name]);
+    } else {
+      displayName = loc(`socialauth.${type}.label`, 'login');
+    }
+
+    const classNames = [
+      'social-auth-button',
+      `social-auth-${type}-button`,
+    ];
+
+    if (type === 'general-idp') {
+      classNames.push('no-translate');
+    }
+
+    // @ts-expect-error OKTA-609464 - className missing from IdpConfig type
+    if (idpObject.idp?.className) {
+      // @ts-expect-error OKTA-609464 - className missing from IdpConfig type
+      classNames.push(idpObject.idp.className);
+    }
+
+    return {
+      type: 'Button',
+      label: displayName,
+      options: {
+        type: ButtonType.BUTTON,
+        step: IDX_STEP.PIV_IDP,
+        dataSe: 'piv-card-button',
+        classes: classNames.join(' '),
+        variant: 'secondary',
+        Icon: getIdpButtonIcon(type),
+        iconAlt: loc('piv.card', 'login'),
```

Comment:
`iconAlt` here needs to match the icon that is actually being used. 

I think the `getIdpButtonIcon(type)` could possibly return an object that contains the Icon as well as the `alt` text. One issue I see though is that the button label cannot be the same as the icon alt text (there is a explicit a11y rule against this). Looking into login.properties I don't see any keys specifically for these social auth icons that can be used (that isnt the same as the label) so you may have to create some like 

```
socialauth.facebook.icon.alt=Facebook
socialauth.google.icon.alt=Google
...
```

Reply:
An alternative to the above if you dont want to create alt text for each of these icons, is you could just leave it as an empty string since the icon on the button uses aria-hidden. The only downside to that is when you hover over the image it wont display alternative text but that may be acceptable since the button label will exist. 

I'll leave that up to you. 🙏

Reply:
Oops my bad this was supposed to be up above for the piv button lol.

I think since the button uses aria-hidden its probably fine to leave it as an empty string